### PR TITLE
Update links in eslint-parser README

### PR DESCRIPTION
**Pull request by codyatwork**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | None 
| Patch: Bug Fix?          | No
| Major: Breaking Change?  | No
| Minor: New Feature?      | No
| Tests Added + Pass?      | No
| Documentation PR Link    | N/A
| Any Dependency Changes?  | No
| License                  | MIT

Updated some outdated links to the Slack invite and eslint.org in the eslint-parser README.

### CODE REVIEW

**Review by JLHwung — APPROVED** [review]
Thanks!

File: eslint/babel-eslint-parser/README.md [hunk 1]
```diff
@@ -105,4 +105,4 @@ While [`@babel/eslint-parser`](https://github.com/babel/babel/tree/main/eslint/b
 
 If you have an issue, please first check if it can be reproduced with the default parser and with the latest versions of `eslint` and `@babel/eslint-parser`. If it is not reproducible with the default parser, it is most likely an issue with `@babel/eslint-parser`.
 
-For questions and support please visit the [`#discussion`](https://babeljs.slack.com/messages/discussion/) Babel Slack channel (sign up [here](https://github.com/babel/notes/issues/38)) or the ESLint [Gitter](https://gitter.im/eslint/eslint).
+For questions and support please visit the [`#discussion`](https://babeljs.slack.com/messages/discussion/) Babel Slack channel (sign up [here](https://slack.babeljs.io/)) or the ESLint [Gitter](https://gitter.im/eslint/eslint).
```
Comment (kaicataldo) [inline eslint/babel-eslint-parser/README.md:108]:
```suggestion
For questions and support please visit the [`#discussion`](https://babeljs.slack.com/messages/discussion/) Babel Slack channel (sign up [here](https://slack.babeljs.io/)) or the [ESLint Discord server](https://eslint.org/chat).
```

**Review by kaicataldo — CHANGES_REQUESTED** [review]
Thanks for doing this! Can we update the ESLint chat link as well, since they've moved over to Discord?

**Review by kaicataldo — APPROVED** [review]
LGTM, thank you!

**Pull request merged**